### ISSUE TITLE
Definitions of numeric identifiers

### ISSUE THREAD

**Issue by vaitkus** [COMCIFS/Modulated_Structures#14]
Similar to issue #13 data items `_atom_sites_axes.matrix_seq_id` and `_atom_site_fourier_wave_vector.seq_id` are defined as integers in the range of `0:` while some of the items that reference them are instead defined as having the `Code` type. What should the data type of these items be (e.g. `Integer`, `Code`, `Word`)?

**Comment by brantonc** [comment]
I can't imagine why _atom_site_fourier_wave_vector.seq_id would be of any type other than Integer.

**Comment by vaitkus** [comment]
Yes, I can now see that an integer value of `_atom_site_Fourier_wave_vector.seq_id ` is expected by some items from the CIF_MAG dictionary (see PR https://github.com/COMCIFS/magnetic_dic/pull/69). If/when that PR is merged, I will update the `_atom_site_Fourier_wave_vector.seq_id` and linked items accordingly if needed.

**Comment by vaitkus** [comment]
PR #22 restricts the values of `_atom_site_fourier_wave_vector.seq_id` to positive integers as discussed in https://github.com/COMCIFS/magnetic_dic/pull/69#issuecomment-1945363554.

The preferred data type and value range of `_atom_sites_axes.matrix_seq_id` still needs to be agreed upon.

**Comment by vaitkus** [comment]
Merging of PR #30 fixed a lot of things. Two minor inconsistencies remain:
- `_atom_site_fourier_wave_vector.seq_id` is defined as having the range of `0:` while in the magnetic CIF dictionary related items have the range of `1:0` (see discussion PR #22).
- `_atom_sites_axes.matrix_seq_id` is defined as an integer with the range of `0:`, however, all of the examples given in the dictionary start at 1. Should the range be changed to 1?

**Comment by vaitkus** [comment]
The `ATOM_SITES_ORTHO` and `ATOM_SITE_DISPLACE_ORTHO` category usage examples in the dictionary currently do start the numbering of `_atom_site_Fourier_wave_vector.seq_id` from 0, however, in all other examples the numbering starts from 0. Just noting this as a reminder to update the examples in case the range is changed.

**Comment by gmadaria** [comment]
The consideration of a zero-order harmonic was necessary in older refinement programs for modulated structures (REMOS, for example), where a fixed basic structure was used and homogeneous displacements were refined. This is no longer used today, and in practice, average coordinates are refined directly. The case of orthogonal functions (now almost obsolete) is different. Orthogonal functions are constructed using a fixed linear combination of trigonometric functions, which may include a homogeneous term. Only the coefficients of these orthogonalized functions are refined, making the zero wave vector important. Should we start all numerical identifiers from 1? It is just a No problem,  it is just a consensus decision.

**Comment by brantonc** [comment]
Hi Gotzon,

I feel the contribution associated with k = 0 should be indexed as 0, the fundamental harmonic should be indexed as 1, and so on.  When k=0 is included in the list, the index starts at 0.  When k=0 is not included in the list, the index starts at 1.

Branton

**Comment by gmadaria** [comment]
Hi Branton,

Hi Branton,

I am not a big fan of these kinds of conventions (when k=0 is included in the list, the index starts at 0; when k=0 is not included, the index starts at 1). It works well for (disciplined) people, but it is prone to errors. There is a tendency to manipulate the CIFs generated by programs, and sometimes it becomes a nightmare for programs that deal with CIFs (validators, for example). For me, and this may be debatable, the identifier codes are simply pointers, without any specific meaning, so the pointer for k=0 could be 0, 1, or any alphanumeric string if it is decided to be. I want to emphasize that this is just my opinion, and I'm open to any suggestions on the matter.

**Comment by brantonc** [comment]
Hi Gotzon,

If I were forced to choose between starting at 0 or 1, I would start at 1.  But I guess it doesn’t really matter.

Best wishes,
Branton
